### ISSUE TITLE
fsevent_watch abnormal termination fails silently, burns cpu

### ISSUE THREAD

**Issue by thefloweringash** [thibaudgg/rb-fsevent#80]
In `irb`, run the example in a thread.

```ruby
require 'rb-fsevent'

fsevent = FSEvent.new
fsevent.watch Dir.pwd do |directories|
  puts "Detected change inside: #{directories.inspect}"
end
Thread.new { fsevent.run }
```

Then abnormally terminate `fsevent_watch`. (Press C-c, or `pkill fsevent_watch`, or attach a debugger to `fsevent_watch`, etc)

At this point, there's a zombie process:
```
76993 s012  Z+     0:00.00 (fsevent_watch)
```

And the ruby interpreter is looping as hard as it can reading nothing from the pipe which is at eof https://github.com/thibaudgg/rb-fsevent/blob/0b8b7367a392d9957344f43d9ed125f011a64e5d/lib/rb-fsevent/fsevent.rb#L55

Ruby: ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
Version: rb-fsevent-0.10.2

**Comment by saluzafa** [comment]
I managed to fix it, please see pull request #81 or https://github.com/saluzafa/rb-fsevent

**Comment by thibaudgg** [comment]
Closed by #81
